src/nccl_ofi_rdma.c

```diff
@@ -5348,9 +5334,18 @@ static int connect(nccl_net_ofi_ep_t *base_ep,
 	case COMM_RECV_CONN:
 		/* COMM_RECV_CONN: Receive connect response message from remote */
 
+recv_conn_resp_retry:
 		ret = post_recv_conn_resp(s_comm, device, ep);
 		if (ret == -FI_EAGAIN) {
-			return 0;
+			/* Since the connect request was sent, we must block for completion here.
```

Comment:
I don't follow this comment.  You always send a connect request, so what are you trying to say?